### DISCUSSION TITLE
Obdlink SX COM port Problem

### DISCUSSION THREAD

**Discussion by EricTCartman** [cedricp/ddt4all#1232] [Q&A]
hello guys and thanks for the hard work :)

i have a problem with ddt4all and obdlink sx cable it does not connect to the car (laguna 3)
i want to mention that renolink app works just fine without problems but not for ddt4all. 
sometimes it would give me permission error could not open port 'COM4': PermissionError(13, 'Access is denied.', None, 5)
and other times it would do this
Initilizing ELM with speed 115200...
Trying to open portCOM4 @ 115200
Checking port speed: 500000
ELM not responding
Port closed
Trying to open portCOM4 @ 38400
Checking port speed: 500000
ELM not responding
Port closed
Trying to open portCOM4 @ 115200
Checking port speed: 500000
ELM not responding
Port closed
Trying to open portCOM4 @ 230400
Checking port speed: 500000
ELM not responding
Port closed
Trying to open portCOM4 @ 57600
Checking port speed: 500000
ELM not responding
Port closed
Trying to open portCOM4 @ 9600
Checking port speed: 500000
ELM not responding
Port closed
Trying to open portCOM4 @ 500000
Checking port speed: 500000
ELM not responding
Port closed

i tried it connect to the car too and same responses, all drivers installed FTDI. any help is welcomed. Thanks so much!

**Comment by KarelSvo** [comment]
Give me your used elm.py file to change.

**Reply by ovidiu2106** [reply]
![20250206_193951](https://github.com/user-attachments/assets/ce6ad6d9-205b-4590-9607-d77ef0cc87b7)

**Reply by Furtif** [reply]
try runs ddt4all as admin

**Reply by ovidiu2106** [reply]
I tried
 it doesn't connect
![Uploading 20250206_163948.jpg…]()

**Reply by Furtif** [reply]
you have selected bad COM maybe

**Reply by ovidiu2106** [reply]
No
![Uploading 20250206_201514.jpg…]()

**Comment by benderfuturama** [comment]
You need to change baudrate to 115200, renolink is using 1000000.

**Reply by ovidiu2106** [reply]
please, can you help me? I bought another 2.1 interface and it still doesn't work It works on the renolink program

**Comment by KarelSvo** [comment]
Simply add the missing baud rates in elm.py ;-)

**Reply by ovidiu2106** [reply]
I have the same problem as the colleague who opened the topic

**Reply by ovidiu2106** [reply]
please, can you help me? I bought another 2.1 interface and it still doesn't work It works on the renolink program

**Reply by ovidiu2106** [reply]
![Uploading 17485350313095766326456492596363.jpg…]()

**Comment by benderfuturama** [comment]
Will all functions of ddt4all work? Maybe they require 115200?

**Comment by Furtif** [comment]
https://github.com/cedricp/ddt4all/issues/836

**Comment by dido17diag** [comment]
have a problem with ddt4all and obdlink sx cable it does not connect

**Comment by Duke-Jones** [comment]
> have a problem with ddt4all and obdlink sx cable it does not connect

Try to do a factory reset: 
[https://support.obdlink.com/support/solutions/articles/43000721545-restore-obdlink-wired-adapter-to-factory-settings](https://support.obdlink.com/support/solutions/articles/43000721545-restore-obdlink-wired-adapter-to-factory-settings)

It has helped me

**Comment by ovidiu2106** [comment]
> salut baieti si multumesc pentru munca grea :)
> 
> Am o problemă cu cablul ddt4all și obdlink sx nu se conectează la mașină (laguna 3) vreau să menționez că aplicația renolink funcționează bine fără probleme, dar nu și pentru ddt4all. uneori mi-ar da o eroare de permisiune nu a putut deschide portul „COM4”: PermissionError(13, „Accesul este refuzat.”, None, 5) și alteori ar face acest lucru . Se verifică viteza portului: 500000 ELM nu răspunde Portul închis Încercăm să deschid portulCOM4 @ 115200 Se verifică viteza portului : 500000 ELM nu răspunde Portul închis Încercăm să deschidem portCOM4 @ 230400 Se verifică viteza portului: 500000 ELM nu răspunde Portul închis @0 Se încearcă deschiderea portului: 57604 500000 ELM nu răspunde Portul închis Încerc să deschid portulCOM4 @ 9600 Se verifică viteza portului: 500000 ELM nu răspunde Portul închis Încerc să deschid portCOM4 @ 500000 Se verifică viteza portului: 500000 ELM nu răspunde Portul închis
> 
> Am încercat să-l conectez și la mașină și aceleași răspunsuri, toți driverele au instalat FTDI. orice ajutor este binevenit. Multumesc mult!

did you solve it?

**Reply by ayllon306** [reply]
add lines speed baud  file    elm.py
clone  config baud and add  config   baud vv1000000

full ok ddt4all dacia 

for s in [38400, 115200, 230400, 57600, 9600, 1000000, 500000]:
            print "\r\t\t\t\t\r" + _("Checking port speed:"), s,
            sys.stdout.flush()



  if baudrate == 38400:
            self.write("at brd 68\r")
        elif baudrate == 57600:
            self.write("at brd 45\r")
        elif baudrate == 115200:
            self.write("at brd 23\r")
        elif baudrate == 230400:
            self.write("at brd 11\r")
        elif baudrate == 500000:
            self.write("at brd 8\r")
        elif baudrate == 1000000:
            self.write("at brd 4\r")         


    def __init__(self, portName, speed, isels = False):
        for s in [int(speed), 38400, 115200, 230400, 57600, 9600, 1000000, 500000]:
            print _("Trying to open port") + "%s @ %i" % (portName, s)
            self.sim_mode = options.simulation_mode
            self.portName = portName
            self.isels = isels




   self.hdr.timeout = 1
        if baudrate == 38400:
            self.hdr.baudrate = 38400
        elif baudrate == 57600:
            self.hdr.baudrate = 57600
        elif baudrate == 115200:
            self.hdr.baudrate = 115200
        elif baudrate == 230400:
            self.hdr.baudrate = 230400
        elif baudrate == 500000:
            self.hdr.baudrate = 500000
        elif baudrate == 1000000:
            self.hdr.baudrate = 1000000

**Reply by ovidiu2106** [reply]
Can you make a text with just what I need to add?

**Reply by ovidiu2106** [reply]
Do I need to delete something or just add something?

**Reply by Mychal7** [reply]
@ayllon306 
Everything works great, I just want to ask if it would be possible to change the default value from 115200 to 1000000 for OBDLINK somewhere
I didn't find it in any file but I guess I was just looking wrong :D
